### PR TITLE
make data-pid & data-bcid mandatory for amp-o2-player extension

### LINKED ISSUE #4538: Make data-pid&data-bcid mandatory for amp-o2-player extension
The following use cases of <amp-o2-player> extension should be valid:
1) data-pid & data-bcid
2) data-pid & data-bcid & data-vid
3) data-pid & data-bcid & data-bid
4) data-pid & data-bcid & data-bid & data-vid

### PR DESCRIPTION
**Pull request by yevheniiminin**
Resolves #4538

### CODE REVIEW

File: extensions/amp-o2-player/0.1/test/test-amp-o2-player.js [hunk 1]
```diff
@@ -69,50 +69,55 @@ describe('amp-o2-player', () => {
     });
   });
 
-  it('requires data-pid && data-bcid or data-vid', () => {
+  it('requires data-pid && data-bcid', () => {
```
Comment (jridgewell) [inline extensions/amp-o2-player/0.1/test/test-amp-o2-player.js]: Please add tests for only a `data-pid` and only a `data-bcid` as well.

File: extensions/amp-o2-player/0.1/validator-amp-o2-player.protoascii [hunk 2]
```diff
@@ -50,18 +50,8 @@ tags: {  # <amp-o2-player>
...
-  }
-  attrs: {
-    name: "data-bcid"
-    mandatory_oneof: "data-vid or data-pid and data-bcid"
-    trigger: {
-      also_requires_attr: "data-pid"
-    }
-  }
+  attrs: { name: "data-pid" mandatory: true }
+  attrs: { name: "data-bcid" mandatory: true }
```
Comment (jridgewell) [inline extensions/amp-o2-player/0.1/validator-amp-o2-player.protoascii]: We need to add the `bid` and `vid` attrs.

**Comment by adylevy** [comment]
adding @raphaeleidus as a watcher

**Comment by yevheniiminin** [comment]
@jridgewell Fixed comments and also fixed an issue with undefined domain_, could you please take a look once again? Thanks!

File: extensions/amp-o2-player/0.1/amp-o2-player.js [hunk 3]
```diff
@@ -31,28 +31,41 @@ class AmpO2Player extends AMP.BaseElement {
...
     /** @private {string} */
     this.domain_ = 'https://delivery.' +
-      (env != 'stage' ? '' : 'dev.') + 'vidible.tv';
+        (env != 'stage' ? '' : 'dev.') + 'vidible.tv';
     let src = `${this.domain_}/htmlembed/`;
     const queryParams = [];
-    if (pid && bcid) {
-      src += 'pid=' + encodeURIComponent(pid) + '/'
-        + encodeURIComponent(bcid) + '.html';
+    if (this.pid_ && this.bcid_) {
```
Comment (jridgewell) [inline extensions/amp-o2-player/0.1/amp-o2-player.js]: This conditional is always true now, since we assert that we have `pid_` and `bcid_` above.

**Comment by jridgewell** [comment]
Fixes https://github.com/ampproject/amphtml/issues/4557.

File: extensions/amp-o2-player/0.1/amp-o2-player.js (hunk 3 above)
Comment (yevheniiminin) [inline extensions/amp-o2-player/0.1/amp-o2-player.js]: Fixed!

**Comment by jridgewell** [comment]
LGTM.

**Pull request merged**